addons/microsoft_calendar/models/microsoft_sync.py

```diff
@@ -74,12 +74,14 @@ def write(self, vals):
 
         result = super().write(vals)
         for record in self.filtered('need_sync_m'):
-            if record.microsoft_id and fields_to_sync:
+            if 'active' in vals.keys() and not vals.get('active') and record.microsoft_id:
```

Comment:
Could take the check on active outside the loop, especially if it's writing on a lot of records at once